properties/base/base.py

```diff
@@ -436,12 +436,51 @@ def equal(self, other):
         Equivalence is determined by checking if all Property values on
         two instances are equal, using :code:`Property.equal`.
         """
-        if self is other:
-            return True
-        if not isinstance(other, self.__class__):
-            return False
-        for prop in itervalues(self._props):
-            if prop.equal(getattr(self, prop.name), getattr(other, prop.name)):
-                continue
-            return False
+        warn('HasProperties.equal has been depricated in favor of '
+             'properties.equal and will be removed in the next release',
+             FutureWarning)
+        return equal(self, other)
+
+
+@utils.stop_recursion_with(False)
+def equal(value_a, value_b):
+    """Determine if two **HasProperties** instances are equivalent
+
+    Equivalence is determined by checking if (1) the two instances are
+    the same class and (2) all Property values on two instances are
+    equal, using :code:`Property.equal`. If the two values are the same
+    HasProperties instance (eg. :code:`value_a is value_b`) this method
+    returns True. Finally, if either value is not a HasProperties
+    instance, equality is simply checked with ==.
+    """
+    if (
+            not isinstance(value_a, HasProperties) or
+            not isinstance(value_b, HasProperties)
+    ):
+        return value_a == value_b
+    if value_a is value_b:
         return True
+    if value_a.__class__ is not value_b.__class__:
+        return False
+    for prop in itervalues(value_a._props):
```

Comment:
If we are going down this path, your idea would make sense. Or could do something like
```
def __hash__(self):
    return hash(self.serialize())
```
That way `hash` would have the same assumptions as `serialize`. However, an objects hash shouldn't change over it's lifetime, and this definitely does.

AND, no matter what we do with hash, `hp_a in [hp_b]` will still use `__eq__` therefore be `True` if both objects are equivalent. Especially if you agree that behavior is a no go, I'm inclined to put this to rest and stick with the not-syntactic-sugar-but-behaves-as-expected `properties.equal` over `__eq__`.